lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py

```diff
@@ -251,7 +252,22 @@ def describe_subnets_with_backoff(client, **params):
     return client.describe_subnets(**params)
 
 
-def create_subnet(conn, module, vpc_id, cidr, ipv6_cidr=None, az=None):
+def wait_config(wait_timeout, start_time):
+    remaining_wait_timeout = int(wait_timeout + start_time - time.time())
+    return {'Delay': 5, 'MaxAttempts': remaining_wait_timeout // 5}
+
+
+def handle_waiter(conn, module, waiter_name, params, start_time):
+    params['WaiterConfig'] = wait_config(module.params['wait_timeout'], start_time)
+    try:
+        get_waiter(conn, waiter_name).wait(
+            **params
+        )
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
```

Comment:
I think it'd be best to separately handle `WaiterError` here, so we can provide a message for "timed out waiting for update to [insert what you waited for here]" and another message for "Error XYZ occured while waiting for [whatever]" so that we can see in bug reports whether the wait time is just too short or if there is an error while waiting.